A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_UpperECAM.js

```diff
@@ -2942,24 +2947,37 @@ var A320_Neo_UpperECAM;
                 //Action
                 const action = document.createElement("span");
                 action.className = "Action";
-                let actionText = _item.action;
-                for (let i = 0; i < (19 - _item.name.length - _item.action.length); i++) {
-                    actionText = "." + actionText;
+                let actionText;
+                if (typeof _item.action === 'function') {
+                    actionText = _item.action();
+                } else {
+                    actionText = _item.action;
                 }
-                action.textContent = actionText;
+                action.textContent = this.leftPad(actionText, ".", 19 - _item.name.length);
                 div.appendChild(action);
 
                 //Completed
                 const completed = document.createElement("span");
                 completed.className = "Completed";
-                completed.textContent = " " + _item.completed;
+                if (typeof _item.completed === 'function') {
+                    completed.textContent = " " + _item.completed();
+                } else {
+                    completed.textContent = " " + _item.completed;
+                }
                 div.appendChild(completed);
 
                 div.className = "InfoIndication";
                 div.setAttribute("id", _item.id);
             }
         }
 
+        leftPad(_text, _pad, _length) {
```

Comment:
This really belongs elsewhere (insert NPM joke). I just factored it out of the code above so I could use it in two places. Happy if anyone can suggest a better place for it.